fdbclient/include/fdbclient/FDBTypes.h

```diff
@@ -546,36 +546,31 @@ struct hash<KeyRange> {
 
 enum { invalidVersion = -1, latestVersion = -2, MAX_VERSION = std::numeric_limits<int64_t>::max() };
 
-inline Key keyAfter(const KeyRef& key) {
-	if (key == "\xff\xff"_sr)
-		return key;
-
-	Standalone<StringRef> r;
-	uint8_t* s = new (r.arena()) uint8_t[key.size() + 1];
-	if (key.size() > 0) {
-		memcpy(s, key.begin(), key.size());
-	}
-	s[key.size()] = 0;
-	((StringRef&)r) = StringRef(s, key.size() + 1);
-	return r;
-}
 inline KeyRef keyAfter(const KeyRef& key, Arena& arena) {
-	if (key == "\xff\xff"_sr)
-		return key;
+	if (key == "\xff\xff"_sr) {
+		return KeyRef(arena, key);
```

Comment:
I suppose this is slightly less efficient if you call `keyAfter("\xff\xff"_sr)`, where we should be able to safely return `"\xff\xff"_sr` (but not the argument itself, which was broken as you found). Still, that may not be common enough to matter.

Reply:
Done